### PR TITLE
Fixing the helm templates to support Control Plane SDS (#16466)

### PR DESCRIPTION
**Pull request by JimmyCYJ**
* support control plane SDS
* test control plane SDS
* set env
* template change for SDS control plane
* enable control plane SDS

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by JimmyCYJ** [comment]
manually create a PR that cherry-picks https://github.com/istio/istio/pull/16466

**Comment by myidpt** [comment]
@rlenglet 
The auto-cherrypick failed: https://github.com/istio/istio/pull/16466

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 1]
```diff
@@ -236,6 +236,8 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
+          - name: SDS_ENABLED
```
Comment (rlenglet) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: @JimmyCYJ Please update this PR's description to link to the equivalent PR that updates the new installer. If you won't update the new installer, please explain why.

**Comment by JimmyCYJ** [comment]
The unit tests fail because https://github.com/istio/istio/pull/16462 is cherry-picked into 1.3 branch, which changes auth_golden.json and running_golden.json. I have to update authsds_golden.json and runningsds_golden.json accordingly.

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml (hunk 1 above)
Comment (JimmyCYJ) [reply install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: @myidpt Could you provide any context about equivalent changes to new installer?

**Pull request merged**